Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs

```diff
@@ -1206,9 +1206,16 @@ public override void DrawAt(Vector3 drawLoc, bool flip = false)
             }
             else
             {
+                Quaternion shadowRotation = ExactRotation;
+                Quaternion projectileRotation = DrawRotation;
+                if (def.projectile.spinRate != 0f)
+                {
+                    float num2 = 60f / def.projectile.spinRate;
+                    shadowRotation = projectileRotation = Quaternion.AngleAxis(Find.TickManager.TicksGame % num2 / num2 * 360f, Vector3.up);
```

Comment:
You account for the projectile rotation and the shadow rotation not aligning on 1209 and 1210, but here you force them both to the same value if there is some spin rate.  Should this rotate them each relative to their input value rather than a fixed value based only on tick and spin rate?

Reply:
Fixed